### PR TITLE
CSS for existing views

### LINKED ISSUE #43: CSS for existing views
Since we've got some views created, it's time to start experimenting with CSS to improve the site visuals.

This can be broken up into several PRs (a separate one for `sites`, `water quality`, and `create account/register`) or all bundled into one. While I'd recommend separating the PRs (for easier organization), I'll leave the decision up to whoever wants to tackle this issue. :)

### PR DESCRIPTION
**Pull request by subnomo**
Fixes #43
## Changes in this PR.

- [X] Add [Materialize](http://materializecss.com/) to the project
- [X] Materialize `/streamwebs/sites` page
- [x] Materialize `/streamwebs/sites/<site_slug>` page
- [ ] Materialize datasheets
  - [x] Water Quality
  - [x] Macroinvertebrates
  - [ ] Riparian transect 
- [x] Materialize base template
- [x] Materialize index page
- [x] Materialize login/register
## Testing this PR.

1. Start the docker container with `docker-compose up`
2. Visit http://localhost:8000/streamwebs/, browse the site a little
### Expected Output.

You should see materialized views. This is an example with the `sites` page:

2016-07-12-121003_1196x770_scrot

@osuosl/devs

### CODE REVIEW

File: streamwebs_frontend/streamwebs/templates/streamwebs/sites.html [hunk 1]
```diff
@@ -25,34 +25,55 @@
...
-        {% for site in sites %}
-            <div name="{{site.slug}}" class="search_item {{site.site_type}}" style="display: none">
-                <p name="{{site.slug}}" onclick="submit_search(this)">{{site.name}}</p>
+    <div class="container">
+        <div id="site_search">
+            <div class="row">
+                <nav class="search-nav col s8 cyan">
+                    <div class="nav-wrapper">
+                        <div class="input-field">
+                            <input id="search" type="search" autocomplete="off" oninput="search()">
```
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/templates/streamwebs/sites.html]: Should probably have some placeholder text so the student/user knows what the search is for

**Comment by alxngyn** [comment]
Aside from the minor feedback given, clean foundation design. +1

**Comment by leian7** [comment]
Screenshot looks great, but when I try to build I get this
mat_sites
even after rebuilding the container while on your branch. Did I miss a step?

**Comment by subnomo** [comment]
@leian7 Probably because the old css has been cached in your browser. Try pressing `Ctrl + Shift + R` to refresh the page and cache.

**Comment by leian7** [comment]
@subnomo Thanks, I tried it but it still didn't give me the right thing. It's probably just something on my end, though.

**Comment by subnomo** [comment]
This breaks the water quality page, so until the other pages have been "materialized", it's best to hold off on merging this one.

**Comment by lo9key** [comment]
@subnomo is this a dead PR?  It looks like a nice presentation to me.

**Comment by subnomo** [comment]
@lo9key No, I'm actually just about to add a bunch of stuff :)

**Comment by alxngyn** [comment]
im working on prettifying the index page. I've started by using materialize's demo page as a template and just filled in the rest so far.
Should i continue fleshing out the homepage?

[Screenshot of index.html](http://i.imgur.com/yQRagJv.png)

**Comment by subnomo** [comment]
@alxngyn Looks great! Please do.

**Comment by alxngyn** [comment]
@subnomo good work on the styling
LGTM before it gets too big.
I think we should leave index and login page for another PR

**Comment by alxngyn** [comment]
i finished login and register page

File: streamwebs_frontend/streamwebs/templates/streamwebs/datasheets/water_quality.html [hunk 2]
```diff
@@ -30,313 +34,321 @@
...
+
+            <div class="row">
+                <div class="col s4">
+                    <p>{{ wq_form.fish_present.label }}</p>
+                    <div class="input-field">
+                        <p id="fish_radio">
+                            {% for radio in wq_form.fish_present %}
+                                {{ radio.tag }}
+                                <label for="{{ radio.id_for_label }}">{{ radio.choice_label }}</label>
+                            {% endfor %}
```
Comment (subnomo) [inline streamwebs_frontend/streamwebs/templates/streamwebs/datasheets/water_quality.html]: @leian7 This is how you do the radio boxes, probably similar for checkboxes.

File: streamwebs_frontend/streamwebs/templates/streamwebs/sites.html [hunk 3]
```diff
@@ -27,35 +27,55 @@
...
+                                <a href="site/{{site.site_slug}}" class="collection-item search-item {{site.site_type}} cyan-text hide">{{site.site_name}}</a>
+                            {% endfor %}
+                            </div>
+                        </div>
+                    </div>
+                </nav>
+
+                <div class="cbox col s4 push-s1">
+                    <div>
+                        <input type="checkbox" id="steward_box" name="stewardship" value="steward" onchange="search()" />
```
Comment (athai) [inline streamwebs_frontend/streamwebs/templates/streamwebs/sites.html]: Can you remove the site_type check boxes?

File: streamwebs_frontend/streamwebs/static/streamwebs/js/sites.js [hunk 4]
```diff
@@ -82,29 +75,39 @@ function initialize() {
...
+                otherWindow.close();
             }
-            site.infoWindow.open(map, site.marker);
+
+            infoWindows[this.index].open(map, markerList[this.index]);
         });
 
-        latSum += site.lat;
-        lngSum += site.lng;
+        lngSum += site.lat;
```
Comment (athai) [inline streamwebs_frontend/streamwebs/static/streamwebs/js/sites.js]: I fixed the ``sites`` data migration script, so the coordinates in the db should be correct now. In other words, you don't need to swap them to display map markers. :)

File: streamwebs_frontend/streamwebs/templates/streamwebs/sites.html [hunk 5]
```diff
@@ -27,35 +27,55 @@
...
-        <input type="checkbox" id="salmon_box" name="salmon_watch" value="salmon" onchange="search()" />
-        <label for="available_box">{% trans "Available Project" %}</label>
-        <input type="checkbox" id="available_box" name="available" value="available" onchange="search()" />
-        <div id="search_dropdown">
-        {% for site in sites %}
-            <div name="{{site.slug}}" class="search_item {{site.site_type}}" style="display: none">
-                <p name="{{site.slug}}" onclick="submit_search(this)">{{site.name}}</p>
+                            <div class="search-results collection">
+                            {% for site in sites %}
+                                <a href="site/{{site.site_slug}}" class="collection-item search-item {{site.site_type}} cyan-text hide">{{site.site_name}}</a>
```
Comment (athai) [inline streamwebs_frontend/streamwebs/templates/streamwebs/sites.html]: Fix link.

File: streamwebs_frontend/streamwebs/templates/streamwebs/sites.html [hunk 6]
```diff
@@ -27,35 +27,55 @@
...
-        {% for site in sites %}
-            <option value="{{site.slug}}">{{site.name}}</option>
-        {% endfor %}
-        </select>
-        <input type="submit" name="Submit" class="Submit" value="{% trans "Select" %}" onclick="submit_select()" />
+        <div id="site_select">
+            <a href="#!" class="dropdown-button btn select-button cyan" data-activates='dropdown'>{% trans "Select Site" %}<i class="material-icons right">arrow_drop_down</i></a>
+            <ul id="dropdown" class="dropdown-content">
+            {% for site in sites %}
+                <li><a href="site/{{site.site_slug}}" class="cyan-text">{{site.site_name}}</a></li>
```
Comment (athai) [inline streamwebs_frontend/streamwebs/templates/streamwebs/sites.html]: Fix this link too.

File: streamwebs_frontend/streamwebs/static/streamwebs/js/sites.js [hunk 7]
```diff
@@ -82,29 +75,39 @@ function initialize() {
...
+    var markerList = [];
+    var infoWindows = [];
+
+    for (var i = 0; i < sites.length; i++) {
+        var site = sites[i];
+
+        markerList[i] = new google.maps.Marker({
             map: map,
-            position: new google.maps.LatLng(site.lat, site.lng),
+            position: new google.maps.LatLng(site.lng, site.lat),
```
Comment (athai) [inline streamwebs_frontend/streamwebs/static/streamwebs/js/sites.js]: You don't need to swap the coordinates anymore.

**Comment by leian7** [comment]
Here's macroinvertebrates: https://github.com/osuosl/streamwebs/pull/162

**Comment by subnomo** [comment]
If anyone has input on what you think `site_detail.html` (visit `/streamwebs/<any site slug>`) should look like, let me know. I added a container to keep it from being aligned on the left side but I'm not really sure how to make it any prettier.

**Comment by athai** [comment]
I think it looks pretty good, but it seems like the navigation bar disappeared? Or did you remove it?  

As for ``site_detail``, I don't think you need to worry about it for now; that can be prettied up later.

**Pull request merged**